modules/pre-commit.nix

```diff
@@ -315,7 +329,43 @@ in
                 echo 1>&2 "    3. add .pre-commit-config.yaml to .gitignore"
               else
                 ln -s ${configFile} .pre-commit-config.yaml
-                pre-commit install
+                # Remove any previously installed hooks (since pre-commit itself has no convergent design)
+                hooks="pre-commit pre-merge-commit pre-push prepare-commit-msg commit-msg post-checkout post-commit"
+                uninstall_help=$(pre-commit uninstall --help)
+                install_help=$(pre-commit install --help)
+                for hook in $hooks; do
+                  # different version of pre-commit support different hooks
+                  if [[ "$uninstall_help" =~ "$hook" ]]; then
+                    pre-commit uninstall -t $hook
+                  fi
+                done
+                # Add hooks for configured stages (only) ...
+                if [ ! -z "${concatStringsSep " " cfg.default_stages}" ]; then
+                  for stage in ${concatStringsSep " " cfg.default_stages}; do
+                    if [[ "$stage" == "manual" ]]; then
+                      continue
+                    fi
+                    # different version of pre-commit support different hooks
+                    if [[ ! "$install_help" =~ "$stage" ]];then
+                        echo 1>&2 "WARNING: nix-pre-commit-hooks: '$stage' is not supported by this pre-commit version."
+                        continue
+                    fi
+                    case $stage in
+                      commit | merge-commit | push)
+                        stage="pre-"$stage
+                        pre-commit install -t $stage
+                        ;;
+                      prepare-commit-msg | commit-msg | post-checkout | post-commit)
+                        pre-commit install -t $stage
+                        ;;
+                      *)
+                        ;;
```

Comment:
iiuc at this point we're dealing with a `$stage` that's probably valid, according to the check with `$install_help =~ $stage`. Instead of doing nothing and failing to install silently, we should allow `pre-commit install -t $stage` to try, or fail with a message that the `$stage` hook is not supported yet.

I think that's the only remaining issue now. Thanks for being patient :)

Reply:
> Thanks for being patient

No worries! I'm aware this is done once and used many.

Since in my last commit I removed the _shady_ regex on the help command, I think you are right.

I'll do the following:
- keep the switch  `prepare-commit-msg | commit-msg | post-checkout | post-commit)` - since we only can uninstall so many (they have to be in $hook`)
- add an error message to `*)`

Reply:
I'll furthermore exit with 1, since we should care about expectations matching the environment.